messaging/src/main/java/org/axonframework/eventhandling/async/SequencingPolicy.java

```diff
@@ -38,15 +40,16 @@
  * @author Allard Buijze
  * @since 0.3
  */
+@FunctionalInterface
 public interface SequencingPolicy<T> {
 
     /**
      * Returns the sequence identifier for the given {@code event}. When two events have the same identifier (as defined
-     * by their equals method), they will be executed sequentially. A {@code null} value indicates that there are no
-     * sequencing requirements for the handling of this event.
+     * by their equals method), they will be executed sequentially. A {@code Optional#empty()} value indicates that
+     * there are no sequencing requirements for the handling of this event.
      *
-     * @param event the event for which to get the sequencing identifier
-     * @return a sequence identifier for the given event
+     * @param event The event for which to get the sequencing identifier.
+     * @return A sequence identifier for the given event.
      */
-    Object getSequenceIdentifierFor(@Nonnull T event);
+    Optional<Object> getSequenceIdentifierFor(@Nonnull T event);
```

Comment:
I don't think this shuld be optional. Any event has a sequence.

Reply:
Let me do the `Also, I think we might want to adjust T to be of EventMessage<?>` switch in the following PRs, because I already have some changes based on it. I've added it to the task #3098 
1. Do I understand correctly that we can get rid of the generics totally from the class?
2. What do you think about the change of the name also? From `SequencingPolicy` to `EventSequencingPolicy`?

Reply:
> Look at other parts, which also operate on "events":

I don't think that's a strong argument to be honest.

> With SequencingPolicy (currently) the T generic is always EventMessage, then payload type is defined in PropertySequencingPolicy, because this one take into account the payload.

I have made `SequencingPolicy<EventMessage<? extends MyCommonEventInterface>>` policies in the past, before the `PropertySequencingPolicy` existed.

--- 

That said, times changes. So if you feel strongly to remove it, go for it! It's a detail with limited use anyhow.

Reply:
Change introduced, I think the PR is ready.